### ISSUE TITLE
[BUG] Pods unable to mount even though Longhorn says it's attached

### ISSUE THREAD

**Issue by jabbermouth** [longhorn/longhorn#2556]
**Describe the bug**
If a pod is restarted, the pod often fails to mount the volume.  Longhorn's UI reports the volume as attached but pending to the same node as the pod trying to mount the volume but the pod remains stuck at ContainerCreating.  This problem only started happening after upgrading to Longhorn 1.1.1 from 1.1.0.

**To Reproduce**
Rebooting a node without cordoning usually leads to at least one pod having issues mounting storage.  Often manually deleting a pod still results in a pod unable to mount the volume.

**Expected behavior**
The volume mounts in <60 seconds and pod successfully starts.

**Log**
This is the events section from the pod:
```
  Type     Reason       Age   From               Message
  ----     ------       ----  ----               -------
  Normal   Scheduled    91s   default-scheduler  Successfully assigned development/bioprotectionportalapp-wordpress-7d679df8cd-d5z8d to kubernetes-node-03
  Warning  FailedMount  59s   kubelet            MountVolume.SetUp failed for volume "pvc-d81593c2-e00f-4cd4-94d0-8fe7ba14a0c4" : rpc error: code = Internal desc = mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t ext4 -o defaults /dev/longhorn/pvc-d81593c2-e00f-4cd4-94d0-8fe7ba14a0c4 /var/lib/kubelet/pods/11f3e6f3-64dc-493e-b4c5-685aa4b08a99/volumes/kubernetes.io~csi/pvc-d81593c2-e00f-4cd4-94d0-8fe7ba14a0c4/mount
Output: mount: /var/lib/kubelet/pods/11f3e6f3-64dc-493e-b4c5-685aa4b08a99/volumes/kubernetes.io~csi/pvc-d81593c2-e00f-4cd4-94d0-8fe7ba14a0c4/mount: /dev/longhorn/pvc-d81593c2-e00f-4cd4-94d0-8fe7ba14a0c4 already mounted or mount point busy.
```

**Environment:**
 - Longhorn version: 1.1.1
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: 'Baremetal' v1.20.2 on Ubuntu v20.04
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 3
 - Node config
   - OS type and version: Ubuntu v20.04
   - CPU per node: 4
   - Memory per node: 8GB
   - Disk type(e.g. SSD/NVMe): SSD
   - Network bandwidth between the nodes: 1 Gbps
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): VMWare
 - Number of Longhorn volumes in the cluster: 20

**Additional context**
Worked absolutely fine on v1.1.0.  Upgraded to v1.1.1 and then problems started.  Nodes have all been patched and upgrade with OS updates.

**Comment by jabbermouth** [comment]
I've just updated to Kubernetes v1.21.0 and still get the same problem.  I get the problem with RWO and RWX volumes.

**Comment by jabbermouth** [comment]
I've noticed that at least sometimes stuff isn't working, the share-manager-pvc pod isn't ready but is running.

```
~$ kubectl describe pod -n longhorn-system share-manager-pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c
Name:         share-manager-pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c
Namespace:    longhorn-system
Priority:     0
Node:         kubernetes-node-02/10.10.4.31
Start Time:   Sun, 02 May 2021 18:37:42 +0000
Labels:       longhorn.io/component=share-manager
              longhorn.io/managed-by=longhorn-manager
              longhorn.io/share-manager=pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c
              longhorn.io/share-manager-image=smi-cd07eea3
Annotations:  longhorn.io/last-applied-tolerations: []
Status:       Running
IP:           10.244.2.39
IPs:
  IP:           10.244.2.39
Controlled By:  ShareManager/pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c
Containers:
  share-manager:
    Container ID:  docker://2414522d9624d66f0ec7aadd097d7961284dff96c21a68949c5db1293040a211
    Image:         longhornio/longhorn-share-manager:v1_20210416
    Image ID:      docker-pullable://longhornio/longhorn-share-manager@sha256:c0528766fa0fd8c21632e76b0a79844fa484b3f58c53c855ac4575f4076abad1
    Port:          <none>
    Host Port:     <none>
    Args:
      --debug
      daemon
      --volume
      pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c
    State:          Running
      Started:      Sun, 02 May 2021 18:37:43 +0000
    Ready:          False
    Restart Count:  0
    Readiness:      exec [cat /var/run/ganesha.pid] delay=3s timeout=1s period=5s #success=1 #failure=3
    Environment:    <none>
    Mounts:
      /host/dev from dev (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-ljx4p (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  dev:
    Type:          HostPath (bare host directory volume)
    Path:          /dev
    HostPathType:
  kube-api-access-ljx4p:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age               From     Message
  ----     ------     ----              ----     -------
  Normal   Pulled     48s               kubelet  Container image "longhornio/longhorn-share-manager:v1_20210416" already present on machine
  Normal   Created    48s               kubelet  Created container share-manager
  Normal   Started    48s               kubelet  Started container share-manager
  Warning  Unhealthy  4s (x9 over 44s)  kubelet  Readiness probe failed: cat: /var/run/ganesha.pid: No such file or directory
```

**Comment by jabbermouth** [comment]
The log contains a lot of the following messages:
```
E0502 18:41:18.508590       1 mount_linux.go:140] Mount failed: exit status 32
Mounting command: mount
Mounting arguments: -t ext4 -o defaults /host/dev/longhorn/pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c /export/pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c
Output: mount: /export/pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c: /host/dev/longhorn/pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c already mounted or mount point busy.

time="2021-05-02T18:41:18Z" level=warning msg="waiting with nfs server start, failed to mount volume" volume=pvc-01eb74b6-7d5a-4a79-853a-3446dc85624c
```

When I upgraded the Longhorn engines from v1.1.0 to v1.1.1, some of the upgrades got stalled, requiring the pods to be restarted to clear it.  I'm wondering if this is causing the issue.  The problem isn't only on one node and eventually all the pods can mount storage so the data is there.

**Comment by joshimoo** [comment]
Can you check if one of the OS/updates enabled multipath.
https://longhorn.io/kb/troubleshooting-volume-with-multipath/

**Comment by jabbermouth** [comment]
Thanks @joshimoo - I applied the suggested change and almost immediately 4 of 7 stuck pods started running.  The remaining problem pods were all on one node so I evicted everything from that node then uncordoned it and two pods came online.  Only 1 is broken (weird not even showing as attached unlike before) so hopefully it's easily sorted.  I'll monitor this for a few days and, if all looks good, close the bug.
